src/lib/yang/alarms.lua

```diff
@@ -249,7 +249,33 @@ function clear_alarm (key)
    end
 end
 
----
+-- Set operator state.
+
+local operator_states = lib.set('none', 'ack', 'closed', 'shelved', 'un-shelved')
+
+function set_operator_state (key, args)
+   assert(args.state and operator_states[args.state],
+          'Not a valid operator state: '..args.state)
+   key = alarm_keys:normalize(key)
+   local alarm = state.alarm_list.alarm[key]
+   if not alarm then
+      -- Return error. Could not locate alarm.
+      return false, 'Error: Set operate state operation failed. Could not locate alarm.'
```

Comment:
Consider raising an error?  You would then get the same kind of result (i.e. bool, error) with a `pcall(set_operator_state, key, args)`.